engine/src/main/java/org/terasology/engine/world/generator/plugin/WorldGeneratorPluginManager.java

```diff
@@ -0,0 +1,87 @@
+// Copyright 2024 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.engine.world.generator.plugin;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.engine.context.Context;
+import org.terasology.engine.core.module.ModuleManager;
+import org.terasology.engine.registry.InjectionHelper;
+import org.terasology.gestalt.assets.ResourceUrn;
+import org.terasology.gestalt.module.Module;
+import org.terasology.gestalt.module.ModuleEnvironment;
+import org.terasology.gestalt.module.dependencyresolution.DependencyResolver;
+import org.terasology.gestalt.module.dependencyresolution.ResolutionResult;
+
+import java.lang.reflect.InvocationTargetException;
+
+public class WorldGeneratorPluginManager {
+
+    private static final Logger logger = LoggerFactory.getLogger(WorldGeneratorPluginManager.class);
+
+    private final Context context;
+
+    public WorldGeneratorPluginManager(Context context) {
+        this.context = context;
```

Comment:
Not sure if the `refresh` part found in `WorldGeneratorManager` was necessary or not. I believe it is used in the menus but not sure if relevant for world generator plugins ?